### ISSUE TITLE
[#3164] QC-Py-32-RL-DQN: deadlock warmup — agent JAMAIS entraine (val_sharpe 0.657 = buy-and-hold degenere)

### ISSUE THREAD

**Issue by jsboige** [jsboige/CoursIA#3341]
## Contexte

Audit de fidelite **#3164** (lot QC-Py-30..35). Finding **REINVENTION involontaire** sur `MyIA.AI.Notebooks/QuantConnect/Python/QC-Py-32-RL-DQN-Trading.ipynb`. **Meme classe de bug que #3326** (PPO actor fige) : un agent RL qui n'apprend jamais, deguise en agent entraine par des outputs flat.

## Bug (deadlock causale, confirme par lecture directe + outputs)

`DQNAgent.total_steps` est incremente **uniquement** dans `update()` (cell `4482cb70`), mais la boucle d'entrainement (cell `9c27dfe8`) ne garde l'appel a `update()` que derriere :

```python
if agent.total_steps > WARMUP_STEPS:   # WARMUP_STEPS = 1000
    loss = agent.update()
```

Comme `total_steps` demarre a 0 (`__init__`) et n'est incremente QUE dans `update()`, la condition `0 > 1000` est **toujours False** -> `update()` n'est jamais appele -> `total_steps` reste 0 pour les 200 episodes. **Boucle causale fermee.**

## Preuves dans les outputs (cell 9c27dfe8, exec)

- `eps=1.000` CONSTANT sur les 200 episodes -> la decroissance epsilon (ligne dans `update()`) n'est jamais executee -> exploration pure permanente.
- `loss=nan` a chaque eval -> `episode_loss` reste vide (`update` jamais appele) -> `np.mean([])` = nan.
- `val_sharpe=+0.657` IDENTIQUE a chaque eval (20,40,...,200) -> le reseau online, fige a l'init, donne le meme argmax deterministe sur le val set.
- Distribution d'actions (cell eval `faeb20e1`) : **Buy: 11278, Hold: 2, Sell: 0, Close: 0** -> le reseau non-entraine output toujours action=1 (Buy). La "strategie" = acheter tout et garder = **buy-and-hold des 15 actions**, pas un apprentissage DQN.
- Le "best model" sauvegarde a l'episode 20 = reseau initial non-entraine.

## Consequence fidelite (#3164)

Le notebook pretend (cell `8a2835f9`) « L'agent apprend une politique d'achat/vente » et (titre) « Entrainer un agent DQN ». En realite l'agent n'apprend **jamais**. Les metriques "Val Sharpe 0.657, Val Return +31.3%, Win Rate 53.19%" sont celles d'un reseau **non-entraine** equivalant a un buy-and-hold, pas le produit d'un apprentissage DQN. Non-fidele au concept DQN (Hasselt 2015 / Mnih DQN requierent des updates TD repetes). Le notebook exporte meme ce modele fige vers QuantConnect ObjectStore pour "deploiement" — un modele non-entraine present comme un modele appris.

## Fix propose (chirurgical, deplace l'increment, restore l'intention)

Idiome standard DQN : `total_steps` compte les **interactions environnement**, incremente a chaque `store_transition`, pas dans `update()`.

Cellule `4482cb70` (DQNAgent), 2 changements :
1. `store_transition` : ajouter `self.total_steps += 1`.
2. `update` : retirer `self.total_steps += 1` (conserver la decroissance epsilon qui y reste).

Ainsi `total_steps` croit (15 transitions/env-step x 15 actifs), depasse 1000 apres ~67 env-steps (debut episode 1), `update()` devient appelable, et le reseau fait enfin `loss.backward(); optimizer.step()`. Epsilon decroit. loss devient un nombre (pas nan). Le DQN apprend rellement.

## Conformite
- Anti-regression : restore l'intention (updates TD), pas un sorry/stub. Conforme (cf #3326).
- C.2/H.3 : re-execution complete apres modif cellule code (GPU, ~10-20 min, yfinance download requis).

See #3164
